python/cugraph/dask/pagerank/pagerank.py

```diff
@@ -0,0 +1,253 @@
+    # the goal isto maximize reuse while minimizing the number of threads.
+    # We want to limit the number of threads to O(len(devices)) and want to
+    # avoid having if be O(len(ipcs)) at all costs!
+    device_handle_map = defaultdict(list)
+    [device_handle_map[dev].append((idx, ipc)) for dev, ipc, idx in ipcs]
+
+    open_ipcs = []
+    for dev, ipcs in device_handle_map.items():
+        open_ipcs.append([[dev], new_ipc_thread(ipcs[0][1], dev)])
+
+    alloc_info_src = []
+    alloc_info_dest = []
+    for dev, t in open_ipcs:
+        inf = t.info()
+        for i in range(len(dev)):
+            alloc_info_src.append([get_device_id(dev[i]), inf[0]])
+            alloc_info_dest.append([get_device_id(dev[i]), inf[1]])
+
+    for t in raw_arrs:
+        raw_info = build_alloc_info(t)
+        alloc_info_src.append([get_device_id(t[2]), raw_info[0]])
+        alloc_info_dest.append([get_device_id(t[2]), raw_info[1]])
+
+    alloc_info_src.sort(key=lambda x: x[0])
+    alloc_info_dest.sort(key=lambda x: x[0])
+
+    final_allocs_src = [a for i, a in alloc_info_src]
+    final_allocs_dest = [a for i, a in alloc_info_dest]
+
+    pr = cugraph.mg_pagerank(final_allocs_src,
+                             final_allocs_dest,
+                             alpha, max_iter)
+
+    [t[1].close() for t in open_ipcs]
+    [t[1].join() for t in open_ipcs]
+
+    return pr
+
+
+def pagerank(edge_list, alpha=0.85, max_iter=30):
+    """
+    Find the PageRank values for each vertex in a graph using multiple GPUs.
+    cuGraph computes an approximation of the Pagerank using the power method.
+    The input edge list should be provided in dask-cudf dataframe
+    with one partition per GPU.
+
+    Parameters
+    ----------
+    edge_list : dask_cudf.DataFrame
+       Contain the connectivity information as an edge list
+       Source 'src' and destination 'dst' columns must be of type 'int32'.
+       Edge weights are not used for this algorithm.
+       Indices must be in the range [0, V-1], where V is the global number
+       of vertices.
+    alpha : float
+       The damping factor alpha represents the probability to follow an
+       outgoing edge, standard value is 0.85.
+       Thus, 1.0-alpha is the probability to “teleport” to a random vertex.
+       Alpha should be greater than 0.0 and strictly lower than 1.0.
+    max_iter : int
+       The maximum number of iterations before an answer is returned.
+       If this value is lower or equal to 0 cuGraph will use the default value,
+       which is 30.
+
+    Returns
+    -------
+    PageRank : dask_cudf.DataFrame
+        Dask GPU DataFrame containing two columns of size V: the vertex
+        identifiers and the corresponding PageRank values.
+
+    Examples
+    --------
+    >>> import dask_cugraph.pagerank as dcg
+    >>> chunksize = dcg.get_chunksize(edge_list.csv)
+    >>> ddf_edge_list = dask_cudf.read_csv(edge_list.csv,
```

Comment:
Does this example run? No need to import dask_cudf?

Reply:
No, it is just a snapshot of the relevant part (like in most cugraph doc). I think that is good enough to understand the API. We will have a notebook for a working example.

Reply:
I personally believe giving a running example will be helpful for the audiences who want to try examples (I will be upset if I read a documentation, type what appears there, and see error messages), but I will let you make the verdict.